src/java.base/share/classes/sun/security/ssl/SSLScope.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * Scopes defining different parts of TLS protocol.
+ */
+
+package sun.security.ssl;
+
+public enum SSLScope {
+    // Handshake scope as in signature_algorithms extension.
+    HANDSHAKE,
+    // Certificate scope as in signature_algorithms_cert extension.
+    CERTIFICATE;
+
+    // Note: the SSLScope is case-insensitive.
+    public static SSLScope nameOf(String name) {
```

Comment:
`valueOf` is probably a better name, and used by other Enum classes for this pattern.

Reply:
Sure, that's fine. I was thinking that the name was more consistent with the `Enum.valuesOf` method. But these are names or types rather than values, and this class is internal, so we don't have to follow that convention.